### PR TITLE
release: add TiDB 3.0.17 release notes

### PR DESCRIPTION
**Pull request by uglyengineer**
### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [x] v3.1 (TiDB 3.1 versions)
- [x] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs/pull/3465
- Other reference link(s):

### CODE REVIEW

File: releases/release-3.0.17.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+## 优化
+
++ TiDB
+
+    - 
+
++ TiKV
+
+    - 
+
```
Comment (overvenus) [inline releases/release-3.0.17.md:20]:
```suggestion

+ TiDB-Lightning

    - 废弃 `[black-white-list]` 参数，新增一种更加简单易用的过滤规则。 [#332](https://github.com/pingcap/tidb-lightning/pull/332)

```

File: releases/release-3.0.17.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+## Bug 修复
+
++ TiDB
+
+    - 
+
++ TiKV
+
+    -
```
Comment (overvenus) [inline releases/release-3.0.17.md:29]:
```suggestion

+ TiDB-Lightning

    - 解决 `log-file` 参数不生效的问题。 [#345](https://github.com/pingcap/tidb-lightning/pull/345)

```

File: releases/release-3.0.17.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
++ TiKV
+
+    - 
+
+## Bug 修复
+
++ TiDB
+
+    -
```
Comment (XuHuaiyu) [inline releases/release-3.0.17.md:25]:
```suggestion
    - 消除统计信息反馈对集群的性能影响 [#18770](https://github.com/pingcap/tidb/pull/18770)
    - Limit batch split count for one request [#18694](https://github.com/pingcap/tidb/pull/18694)
    - 加速 HTTP API `/tiflash/replica` 再集群中存在大量历史 DDL 操作记录时的访问速度 [#18386](https://github.com/pingcap/tidb/pull/18386)
    - 提升索引等值条件下的行数估算准确率 [#17609](https://github.com/pingcap/tidb/pull/17609)
    - 加快 `kill tidb conn_id` 的响应速度 [#18506](https://github.com/pingcap/tidb/pull/18506)
```

File: releases/release-3.0.17.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+发版日期：2020 年 8 月 3 日
+
+TiDB 版本：3.0.17
+
+## 优化
+
++ TiDB
+
+    -
```
Comment (XuHuaiyu) [inline releases/release-3.0.17.md:15]:
```suggestion
    - 当一个查询中含有 IndexHashJoin 或 IndexMergeJoin 算子，且该算子的子节点发生 panic 时候，返回客户端 panic 的原因而非返回空结果 [#18498](https://github.com/pingcap/tidb/pull/18498)
    - 修复形如 `select a from t having t.a` 的查询返回 UnknowColumn 错误的问题 [#18432](https://github.com/pingcap/tidb/pull/18432)
    - 当一张表没有主键，或其主键为整型时，禁止在这张表上执行添加主键 [#18342](https://github.com/pingcap/tidb/pull/18342)
    - 对 `explain format="dot" for connection` 始终返回空结果 [#17157](https://github.com/pingcap/tidb/pull/17157)
    - 修复 STR_TO_DATE 函数处理 '%r', '%h' 的行为 [#18725](https://github.com/pingcap/tidb/pull/18725)

```

File: releases/release-3.0.17.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+    - 当一个查询中含有 IndexHashJoin 或 IndexMergeJoin 算子，且该算子的子节点发生 panic 时候，返回客户端 panic 的原因而非返回空结果 [#18498](https://github.com/pingcap/tidb/pull/18498)
+    - 修复形如 `select a from t having t.a` 的查询返回 UnknowColumn 错误的问题 [#18432](https://github.com/pingcap/tidb/pull/18432)
+    - 当一张表没有主键，或其主键为整型时，禁止在这张表上执行添加主键 [#18342](https://github.com/pingcap/tidb/pull/18342)
+    - 对 `explain format="dot" for connection` 始终返回空结果 [#17157](https://github.com/pingcap/tidb/pull/17157)
+    - 修复 STR_TO_DATE 函数处理 '%r', '%h' 的行为 [#18725](https://github.com/pingcap/tidb/pull/18725)
+
+
++ TiKV
+
+    -
```
Comment (ran-huang) [inline releases/release-3.0.17.md:24]: @breeswish Please add TiKV's release notes here. Thanks!

Comment (uglyengineer) [reply releases/release-3.0.17.md]: done

**Comment by XuHuaiyu** [comment]
LGTM

File: releases/release-3.0.17.md [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+## Bug 修复
+
++ TiDB
+
+    - 当一个查询中含有 `IndexHashJoin` 或 `IndexMergeJoin` 算子，且该算子的子节点发生 panic 时，返回客户端 panic 的原因，而非返回空结果 [#18498](https://github.com/pingcap/tidb/pull/18498)
+    - 修复形如 `SELECT a FROM t HAVING t.a` 的查询返回 `UnknowColumn` 错误的问题 [#18432](https://github.com/pingcap/tidb/pull/18432)
+    - 当一张表没有主键，或其主键为整型时，禁止在这张表上执行添加主键 [#18342](https://github.com/pingcap/tidb/pull/18342)
+    - 对 `EXPLAIN FORMAT="dot" FOR CONNECTION` 始终返回空结果 [#17157](https://github.com/pingcap/tidb/pull/17157)
+    - 修复 `STR_TO_DATE` 函数处理 `'%r'`, `'%h'` 的行为 [#18725](https://github.com/pingcap/tidb/pull/18725)
```
Comment (TomShawn) [inline releases/release-3.0.17.md:19]:
```suggestion
    - 修复 `STR_TO_DATE` 函数处理 `'%r'` 和 `'%h'` 的行为 [#18725](https://github.com/pingcap/tidb/pull/18725)
```

File: releases/release-3.0.17.md [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
++ TiDB Lightning
+
+    - 解决 `log-file` 参数不生效的问题 [#345](https://github.com/pingcap/tidb-lightning/pull/345)
+
+## 优化
+
++ TiDB
+
+    - 消除统计信息反馈对集群的性能影响 [#18770](https://github.com/pingcap/tidb/pull/18770)
+    - 限制单次 split 请求中的 region 个数 [#18694](https://github.com/pingcap/tidb/pull/18694)
```
Comment (TomShawn) [inline releases/release-3.0.17.md:34]:
```suggestion
    - 限制单次 split 请求中的 Region 个数 [#18694](https://github.com/pingcap/tidb/pull/18694)
```

File: releases/release-3.0.17.md [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+
+    - 当一个查询中含有 `IndexHashJoin` 或 `IndexMergeJoin` 算子，且该算子的子节点发生 panic 时，返回客户端 panic 的原因，而非返回空结果 [#18498](https://github.com/pingcap/tidb/pull/18498)
+    - 修复形如 `SELECT a FROM t HAVING t.a` 的查询返回 `UnknowColumn` 错误的问题 [#18432](https://github.com/pingcap/tidb/pull/18432)
+    - 当一张表没有主键，或其主键为整型时，禁止在这张表上执行添加主键 [#18342](https://github.com/pingcap/tidb/pull/18342)
+    - 对 `EXPLAIN FORMAT="dot" FOR CONNECTION` 始终返回空结果 [#17157](https://github.com/pingcap/tidb/pull/17157)
+    - 修复 `STR_TO_DATE` 函数处理 `'%r'` 和 `'%h'` 的行为 [#18725](https://github.com/pingcap/tidb/pull/18725)
+
++ TiKV
+
+    - 修复在 Region 合并过程中可能导致读到旧数据的问题 [#8111](https://github.com/tikv/tikv/pull/8111)
```
Comment (breeswish) [inline releases/release-3.0.17.md:23]:
```suggestion
    - 修复在 Region 合并过程中可能导致读到旧数据的问题 [#8111](https://github.com/tikv/tikv/pull/8111)
    - 修复调度时可能产生内存泄漏的问题 [#8355](https://github.com/tikv/tikv/pull/8355)
```

**Pull request merged**